httpserver/mcp_advanced_integration_test.go

```diff
@@ -0,0 +1,521 @@
+//go:build integration
+
+//nolint:errcheck,noctx,gosec,errorlint,govet // Integration test file with acceptable test patterns
+package httpserver
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/bbockelm/golang-htcondor/httpserver"
+	"github.com/bbockelm/golang-htcondor/mcpserver"
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/compose"
+	"github.com/ory/fosite/handler/openid"
+	"github.com/ory/fosite/token/jwt"
+)
+
+// TestDynamicClientRegistration tests OAuth2 dynamic client registration
+func TestDynamicClientRegistration(t *testing.T) {
+	// Skip if condor_master is not available
+	if _, err := exec.LookPath("condor_master"); err != nil {
+		t.Skip("condor_master not found in PATH, skipping integration test")
+	}
+
+	// Setup server (reuse helper from main test)
+	tempDir, server, baseURL, signingKeyPath := setupTestServer(t, 18082)
+	defer os.RemoveAll(tempDir)
+	defer shutdownTestServer(t, server)
+
+	client := &http.Client{Timeout: 30 * time.Second}
+	testUser := "regtest"
+
+	// Test dynamic client registration
+	t.Log("Testing dynamic client registration...")
+
+	regReq := map[string]interface{}{
+		"redirect_uris":  []string{"http://localhost:18082/callback"},
```

Comment:
`setupTestServer` should return the port that the test server started on.  Use that here.

Make corresponding changes to all the unit tests in this file to avoid hardcoding ports.